<?xml version="1.0"?>
<launch>

    <!-- Associate to the robot description parameter, the urdf file that models the robot -->
    <param name="robot_description" command = "$(find xacro)/xacro --inorder $(find pose_estimation)/urdf/ur5_cam.xacro" />
    <!-- Spawn the separate table model -->
    <param name="table_description" command="$(find xacro)/xacro --inorder $(find pose_estimation)/urdf/table.urdf.xacro" />

    <!-- Set the position in empty world -->
    <arg name="x" default="0" />
    <arg name="y" default="0" />
    <arg name="z" default="0" />
    
    <!-- Launch Gazebo world in paused mode -->
    <include file = "$(find gazebo_ros)/launch/empty_world.launch" >
        <arg name="use_sim_time" value="true"/>
        <arg name="debug" value="false"/>
        <arg name="gui" value="true"/>
        <arg name="paused" value="true" />
    </include>   

    <!-- Spawn the robot using the package gazebo_ros -->
    <node name="spawn_the_robot" pkg="gazebo_ros" type="spawn_model" output="screen" 
        args="-urdf -param robot_description -model ur5_robot -x $(arg x) -y $(arg y) -z $(arg z)" />

    

    <!-- Spawn the table using the package gazebo_ros -->
    <node name="spawn_the_table" pkg="gazebo_ros" type="spawn_model" output="screen"
          args="-urdf -param table_description -model my_table -x 1.3 -y 0 -z 0" />


     <group ns="spawn_objects">
        <rosparam file="$(find pose_estimation)/config/spawn_objects.yaml"/>
    </group>
    <node name="spawn_objects_node" pkg="pose_estimation" type="spawn_objects.py" output="screen">
        
    </node>

    
</launch>
